src/reanimated2/component/KeyboardAvoidingView.tsx

```diff
@@ -0,0 +1,142 @@
+import React, { forwardRef, useCallback } from 'react';
+import type {
+  KeyboardAvoidingViewProps,
+  LayoutRectangle,
+  View,
+  ViewProps,
+} from 'react-native';
+import { StyleSheet, useWindowDimensions } from 'react-native';
+import { AnimatedView as ReanimatedView } from './View';
+import {
+  useAnimatedKeyboard,
+  useAnimatedStyle,
+  useWorkletCallback,
+  runOnUI,
+  useSharedValue,
+} from '../';
+
+type Props = ViewProps & KeyboardAvoidingViewProps;
+
+/**
+ * View that moves out of the way when the keyboard appears by automatically
+ * adjusting its height, position, or bottom padding.
+ */
+const KeyboardAvoidingView = forwardRef<View, React.PropsWithChildren<Props>>(
+  (
+    {
+      behavior,
+      children,
+      contentContainerStyle,
+      enabled = true,
+      keyboardVerticalOffset = 0,
+      style,
+      onLayout,
+      ...props
+    },
+    ref
+  ) => {
+    const initialFrame = useSharedValue<LayoutRectangle | null>(null);
+
+    const currentFrame = useSharedValue<LayoutRectangle | null>(null);
+
+    const keyboard = useAnimatedKeyboard();
+    const { height: screenHeight } = useWindowDimensions();
+
+    const onLayoutWorklet = useWorkletCallback((layout: LayoutRectangle) => {
+      if (initialFrame.value == null) {
+        initialFrame.value = layout;
+      }
+
+      currentFrame.value = layout;
+    });
```

Comment:
I suggest that we don't use the _useWorkletCallback_ hook since it has some problems with dependencies. Also, custom hooks don't get eslint suggestions about dependencies.
```suggestion
    const onLayoutWorklet = useCallback(
      (layout: LayoutRectangle) => {
        'worklet';
        if (initialFrame.value == null) {
          initialFrame.value = layout;
        }
        
        currentFrame.value = layout;
      },
      [currentFrame, initialFrame] 
    );
```